resources/knative-eventing/charts/knative-eventing/templates/pre-upgrade-job.yaml

```diff
@@ -36,29 +45,108 @@ data:
   pre-upgrade.sh: |
     #!/usr/bin/env bash
     set -eu
+    RED='\033[0;31m'
+    GREEN='\033[0;32m'
+    NC='\033[0m' # No Color
+
+    function shout() {
+        echo -e "${GREEN}
+    #################################################################################################
+    # $1
+    #################################################################################################
+        ${NC}"
+    }
+
+    function shoutFail() {
+        echo -e "${RED}
+    #################################################################################################
+    # $1
+    #################################################################################################
+        ${NC}"
+    }
+
+    function removeFinalizersFromSubscriptions() {
+      shout "+ Patching all subscriptions"
+      ignoredNamespaces='kyma-system|istio-system|knative-eventing|knative-serving|kube-node-lease|kube-public|kube-system|kyma-installer|natss'
+      for ns in $(kubectl get ns --no-headers | grep -vE "${ignoredNamespaces}" | awk '{print $1}'); do
+         for sub in $(kubectl get -n "${ns}" subscriptions.messaging.knative.dev --ignore-not-found -ojsonpath='{.items[*].metadata.name}'); do
+           if [[ ! -z "${sub}" ]]; then
+             shout "++ Patching subscription: ${sub} in ns: ${ns}"
+             kubectl patch -n "${ns}" subscriptions.messaging.knative.dev "${sub}" --type merge -p '{"metadata": {"finalizers": []}}'
+           fi
+         done
+      done
+    }
+
+    function deleteEventingControllerDeployment() {
+      # This stops reconciling subscriptions
+      shout "+ Deleting eventing-controller"
+      kubectl delete --ignore-not-found -n knative-eventing deploy eventing-controller
+    }
+
+    function waitUntilEventingControllerPodExists() {
+      timeout=120 # seconds
+      pause=0
```

Comment:
I suggest to rename **pause** to something like **elapsed**.

Reply:
changed to `elapsed`